### ISSUE TITLE
Android client after url change: Server not available

### ISSUE THREAD

**Issue by pascalBokBok** [nextcloud/android#13454]
### ⚠️ Before posting ⚠️

- [X] This is a **bug**, not a question or an enhancement.
- [X] I've [searched for similar issues](https://github.com/nextcloud/android/issues) and didn't find a duplicate.
- [X] I've written a clear and descriptive title for this issue, not just "Bug" or "Crash".
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Steps to reproduce

1. move your Nextcloud server folder from domain.com/old-dir to domain.com/new-dir
2. create a 301 permanent redirect from the old folder to the new folder.

### Expected behaviour

The mobile sync client should transition to the new path and keep working. Like the desktop client does.

### Actual behaviour

When opening the mobile nextcloud client, it shows the cached folders and a "server not available" warning. When entering one of the folders it shows loading and the warning about server unavailable.
```
- - [21/Aug/2024:08:43:37 +0000] "GET /owncloud/ocs/v2.php/cloud/user?format=json HTTP/1.1" 301 4473
 - pascal [21/Aug/2024:08:43:37 +0000] "GET /nextcloud/ocs/v2.php/cloud/user?format=json HTTP/1.1" 200 2384
 - - [21/Aug/2024:08:43:37 +0000] "PROPFIND /owncloud/remote.php/dav/files/pascal/Dilbert/ HTTP/1.1" 301 4408
 - pascal [21/Aug/2024:08:43:37 +0000] "PROPFIND /nextcloud/remote.php/dav/files/pascal/Dilbert/ HTTP/1.1" 207 2186
```

### Android version

14

### Device brand and model

Samsung S22

### Stock or custom OS?

Stock

### Nextcloud android app version

3.29.2

### Nextcloud server version

29.0.3

### Using a reverse proxy?

No

**Comment by pascalBokBok** [comment]
I might just add:
When selecting "Activities" in the mobile app it correctly shows the latest activities - way after the server was moved.
And the auto-upload seems halfway working. The auto Image uploads is working, but videos seem to be failing.
It's like the app is connected - but can't read files...

**Comment by pascalBokBok** [comment]
Sharing e.g. a picture to the Nextcloud app also works fine with uploading it. I can even see it in the "Activities" afterwards. 
But listing files seem impossible.

**Comment by GAS85** [comment]
Could be that this bug is known for 5 years? #4157 
Also here some what works and what not: https://github.com/nextcloud/android/issues/4157#issuecomment-666978637

**Comment by pascalBokBok** [comment]
yes indeed. However as written above, it half works. So I'd like to keep this as a bug rather than a feature.

**Comment by ThisIsPaulDaily** [comment]
@joshtrichards It isn't a duplicate because the requested fix is different. 

It is a bug and we should be able to update an account URL from the app. We can do it from a laptop. 

The duplicate ticket wants a creative redirect system. 

A compromise could be to check if the old URL was a trusted host URL of the new one. Though that seems like a bad idea in some ways. 


I really, really don't want to have to delete my account and reconfigure everything on my devices just because the IP changed from a 192.168.0.x to 192.168.1.x
